internal/release/cross/release_list.go

```diff
@@ -147,6 +149,15 @@ func (r *ReleaseList) GetReleasesForPromotion(sourceEnv, targetEnv *v1alpha1.Env
 		// Determine source and target releases
 		sourceRelease := item.Releases[sourceEnvIndex]
 		targetRelease := item.Releases[targetEnvIndex]
+
+		//Check for version in source Release
+		if targetEnv.Name == "qa" || targetEnv.Name == "production" {
+			version := "v" + sourceRelease.Spec.Version
+			if semver.Prerelease(version) != "" || semver.Build(version) != "" {
+				continue
+			}
+		}
```

Comment:
We should never hard-code any specific environment names in Joy, those are nesto specific, and even if Joy was not open source we should never hard-code that kind of logic! 😅  As we discussed yesterday, the check should use the environment's promotion configuration:
```suggestion
		version := "v" + sourceRelease.Spec.Version
		if !targetEnv.Spec.Promotion.FromPullRequests && (semver.Prerelease(version) != "" || semver.Build(version) != "") {
			continue
		}
```